File: single-number/single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int num : nums){
            int count = map.getOrDefault(num, 0);
            map.put(num, count + 1);
        }
        for(int num : nums){
            if(map.get(num) == 1) return num;
        }
        
        return -1;
    }
}

File: single-number/single-number.kt
class Solution {
    fun singleNumber(nums: IntArray): Int {
        
        val map = mutableMapOf<Int, Int>()
        
        nums.forEachIndexed{ index, num ->
            map[num] = map.getOrDefault(num, 0) + 1
        }
        
        for( num in nums ){
            if(map[num] == 1) return num
        }
        
        return -1
        
    }
}